src/main/java/org/apache/cassandra/sidecar/routes/restore/AbortRestoreJobHandler.java

```diff
@@ -69,33 +73,47 @@ protected void handleInternal(RoutingContext context,
                                   HttpServerRequest httpRequest,
                                   String host,
                                   SocketAddress remoteAddress,
-                                  String jobId)
+                                  AbortRestoreJobRequestPayload payload)
     {
         RoutingContextUtils
         .getAsFuture(context, SC_RESTORE_JOB)
-        .compose(job -> {
+        .map(job -> {
             if (RestoreJobStatus.isFinalState(job.status))
             {
-                return Future.failedFuture(wrapHttpException(HttpResponseStatus.CONFLICT,
-                                                             "Job is already in final state: " + job.status));
+                throw wrapHttpException(HttpResponseStatus.CONFLICT,
+                                        "Job is already in final state: " + job.status);
             }
 
-            restoreJobDatabaseAccessor.abort(job.jobId);
+            restoreJobDatabaseAccessor.abort(job.jobId, payload.reason());
+            logger.info("Successfully aborted restore job. job={} remoteAddress={} instance={} reason='{}'",
+                        job, remoteAddress, host, payload.reason());
             restoreJobManagerGroup.signalRefreshRestoreJob();
-            return Future.succeededFuture(job);
+            return job;
         })
         .onSuccess(job -> {
-            logger.info("Successfully aborted restore job. job={}, remoteAddress={}, instance={}",
-                        job, remoteAddress, host);
             metrics.failedJobs.metric.update(1);
             context.response().setStatusCode(HttpResponseStatus.OK.code()).end();
         })
-        .onFailure(cause -> processFailure(cause, context, host, remoteAddress, jobId));
+        .onFailure(cause -> processFailure(cause, context, host, remoteAddress, payload));
     }
 
+    @NotNull
     @Override
-    protected String extractParamsOrThrow(RoutingContext context)
+    protected AbortRestoreJobRequestPayload extractParamsOrThrow(RoutingContext context)
     {
-        return context.pathParam("jobId");
+        String bodyString = context.body().asString(); // nullable
+
+        try
+        {
+            return Json.decodeValue(bodyString, AbortRestoreJobRequestPayload.class);
+        }
+        catch (Exception anyException)
+        {
+            if (bodyString != null)
+            {
+                logger.warn("Failed  to deserialize json string into AbortRestoreJobRequestPayload", anyException);
```

Comment:
Remove extra space here
```suggestion
                logger.warn("Failed to deserialize json string into AbortRestoreJobRequestPayload", anyException);
```